ocpi-toolkit-2.2.1/src/main/kotlin/com/izivia/ocpi/toolkit/common/HttpUtils.kt

```diff
@@ -22,10 +33,10 @@ inline fun <reified T> HttpResponse.parsePaginatedBody(offset: Int): OcpiRespons
         .let { parsedBody ->
             OcpiResponseBody(
                 data = parsedBody.data?.toSearchResult(
-                    totalCount = headers["X-Total-Count"]!!.toInt(),
-                    limit = headers["X-Limit"]!!.toInt(),
+                    totalCount = getHeader(Header.X_TOTAL_COUNT)!!.toInt(),
+                    limit = getHeader(Header.X_LIMIT)!!.toInt(),
```

Comment:
I realize that the use of `!!` will make error reporting very poor if those headers are missing. we should rather raise a custom exception and return the appropriate error response. This is not part of this PR scope, but we should create an issue for that

Reply:
issue created: https://github.com/IZIVIA/ocpi-toolkit/issues/17